Add namespace with a bunch of useful utils for working with 71

aengelberg opened the pull request:
Based on the discussion in #16, it's clear that there is a need for helper functions to simplify common operations with the number 71, but if we go that route, they should be in a separate dedicated namespace. So in this pull request I created `com.gfredericks.seventy-one.util` which has four new exciting functions, including `fast-seventy-one` per Gary's suggestion in #2.

aengelberg committed:
Version bumps

aengelberg committed:
Add com.gfredericks.seventy-one.util with some useful utils

gfredericks commented on src/com/gfredericks/seventy_one/util.cljc:
```diff
@@ -0,0 +1,38 @@
+
+(defn seventy-one?
+  "Checks whether an object is a number that is equal to seventy
+  one. Returns true for integral and floating-point seventy-ones."
```
I imagine it's also true for bigdecs, no?

gfredericks commented on src/com/gfredericks/seventy_one/util.cljc:
```diff
@@ -0,0 +1,38 @@
+     "Threads the expr through the forms. Inserts x as the second item
+  and 71 as the last item in the first form, making a list of it if it
+  is not a list already. If there are more forms, inserts the first
+  form and 71 into the second form, etc."
```
did some use case inspire this particular definition?

aengelberg replied:
Not sure which aspect of this definition you are concerned about, but I can confirm that this macro as written would make a lot of my code more concise. An example:

```clojure
(defn encode-sensitive-data
  "Obfuscate all of my personal data such as passwords, SSN, etc. by
  applying several bit transformations using 71."
  [data]
  (->> (.getBytes data)
       (map (fn [b]
              (->71 b
                    bit-xor
                    bit-shift-left
                    bit-xor
                    bit-shift-left
                    (- 1)
                    bit-shift-right
                    bit-shift-right)))
       (byte-array)))
```

gfredericks replied:
I guess I was wondering why making `71` the last item rather than, e.g., the second, was more useful.

aengelberg replied:
Either of those would work for my use cases; as far as I know, making it the last item just makes me able to assoc 71 into maps as a value to various keys. Making it the second item doesn't seem to gain anything.

gfredericks replied:
what about `->17` threads second, `->>17` threads last?

sometimes you want to assoc something onto a map under the key `71`.

aengelberg replied:
The reason I didn't want to introduce a `->>71` is that the name may mislead the user into thinking that the _form_ gets threaded last, whereas the form is actually still getting threaded first (like `->71`), but the position of `71` is what changes.

gfredericks replied:
maybe we need a more complex naming scheme, since there's a combinatorial issue here.

Possible variants:

- threads form first, 71 second
- threads 71 first, form second
- threads form first, 71 last
- threads 71 first, form last
- threads form last, 71 penultimately
- threads 71 last, form penultimately

Maybe name these `->x71>`, `->71x>`, `->x>71`, `->71>x`, `->>71x`, `->>x71`, respectively? Happy to consider other ideas.

aengelberg replied:
I'm happy to implement all of these (and write test cases for them) if you see value in them, but I'm concerned about introducing unnecessary macros that no one is realistically going to use. This library is widely used enough that it would be worth taking a step back to see which of these really belong in the core API.

gfredericks replied:
Or we could put them in a `.alpha` namespace

gfredericks commented on src/com/gfredericks/seventy_one/util.cljc:
```diff
@@ -0,0 +1,38 @@
+       x)))
+
+(defn gen-seventy-one
+  "Returns a clojure.test.check generator that returns 71. Useful for
```
the second "returns" should be "generates"

gfredericks commented:
Hmm, I just realized we don't have a CI setup. I'll file an issue.

aengelberg committed:
Touch up docstrings per feedback

aengelberg commented:
Closing this since to avoid clutter as I don't plan to work on addressing the code review comments here in the near future, but feel free to re-open if someone else wants to take it over the line.
